### DISCUSSION TITLE
before_commit event but not for nested ones

### DISCUSSION THREAD

**Discussion by kasium** [sqlalchemy/sqlalchemy#11888] [Usage Questions] [answered]
Hi,

I would like that a before_commit event is called but only for "real" commits and not nested ones. Meaning, that currently a before_commit  event is called when a ``session.begin_nested`` block is left even if no "real commit" is executed

**Comment by zzzeek** [comment, marked as answer]
check [session.in_nested_transaction()](https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.in_nested_transaction) for False in your event hook

**Reply by kasium** [reply]
Thanks, I was not aware of this!
